awxkit/awxkit/api/pages/base.py

```diff
@@ -177,6 +174,19 @@ def get_oauth2_token(self, username='', password='', client_id=None, description
         else:
             raise exception_from_status_code(resp.status_code)
 
+    def get_oauth2_token(self, username='', password='', client_id=None, description='AWX CLI', client_secret=None, scope='write'):
+        default_cred = config.credentials.default
+        username = username or default_cred.username
+        password = password or default_cred.password
+        AuthUrls = collections.namedtuple("AuthUrls", ["access_token", "personal_token"])
+        # Try gateway first, fallback to controller
+        try:
+            urls = AuthUrls(access_token="/o/token/", personal_token=f"{config.gateway_base_path}v1/users/")
+            return self._request_token(urls, username, password, client_id, description, client_secret, scope)
```

Comment:
Which one of these actually generates a `NotFound`? It's not clear and may change over time. It is strongly recommended to have only one instruction in try blocks. Could you move one of them out?